src/api/layers/WMSCapabilitiesParser.class.js

```diff
@@ -42,6 +42,43 @@ export default class WMSCapabilitiesParser {
         this.originUrl = new URL(originUrl)
     }
 
+    /**
+     * Returns information about where feature identification can be accessed, or null if this WMS
+     * has no such capabilities
+     *
+     * @param {boolean} ignoreError
+     * @returns {null | ExternalLayerGetFeatureInfoCapability}
+     * @see https://www.mediamaps.ch/ogc/schemas-xsdoc/sld/1.2/capabilities_1_3_0_xsd.html#Capability
+     */
+    getFeatureInfoCapability(ignoreError = true) {
+        if (Array.isArray(this.Capability?.Request?.GetFeatureInfo?.DCPType)) {
+            const httpElement = this.Capability.Request.GetFeatureInfo?.DCPType[0].HTTP
+            let baseUrl = null
+            let method = 'GET'
+            if (httpElement?.Get) {
+                baseUrl = httpElement.Get.OnlineResource
+            } else if (httpElement?.Post) {
+                method = 'POST'
+                baseUrl = httpElement.Post.OnlineResource
+            } else {
+                log.error(
+                    "Couldn't parse GetFeatureInfo data",
+                    this.Capability.Request.GetFeatureInfo
+                )
+                if (ignoreError) {
+                    return null
+                }
+                throw new CapabilitiesError('Invalid GetFeatureInfo data')
+            }
+            return {
+                baseUrl,
+                method,
+                formats: [...this.Capability.Request.GetFeatureInfo.Format],
```

Comment:
What if `Format` is undefined ? Maybe `...this.Capability.Request.GetFeatureInfo.Format ?? {}`

Reply:
As I'm using this array later and wouldn't want "default" value such as an empty Object, I'll check that Format exists before pushing it into the formats array.
This way, the formats array will stay empty if no format is defined by the service